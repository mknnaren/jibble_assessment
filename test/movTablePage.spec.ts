import { mount, shallowMount, createLocalVue  } from '@vue/test-utils'
import movTablePage from '@/components/movie/page.vue';
import Vuetify from 'vuetify';
import Vue from 'vue'
Vue.use(Vuetify)

describe('movTablePage', () => {
  let localVue = createLocalVue()
  let vuetify = new Vuetify()


  test('mounts properly', () => {
    const wrapper = mount(movTablePage, {localVue, vuetify,
      propsData: {
        pageData: {
          page: 1,
          per_page: 10,
          total_pages: 10,
          total: 100
        }
      }})
    expect(wrapper.vm).toBeTruthy()
  })
  test('update pagination triggers properly', async() => {
    const wrapper = mount(movTablePage, {localVue, vuetify,
      propsData: {
        pageData: {
          page: 1,
          per_page: 10,
          total_pages: 10,
          total: 100
        }
      }})
      await wrapper.vm.$nextTick()
      const cb = jest.fn()
      wrapper.vm.$on('input', cb)
      const navigation = wrapper.findAll('.v-pagination__item').wrappers
      navigation[1].trigger('click')
      expect(wrapper.emitted()["show-page-list"]).toBeTruthy()
  })
  test('renders properly', () => {
    const wrapper = mount(movTablePage,  {localVue, vuetify, propsData: {
      pageData: {
        page: 1,
        per_page: 10,
        total_pages: 10,
        total: 100
      }
    }})
    expect(wrapper.html()).toMatchSnapshot()
  })
})
