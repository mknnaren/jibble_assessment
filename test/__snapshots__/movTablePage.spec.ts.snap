// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movTablePage renders properly 1`] = `
"<div>
  <div class=\\"text-center pt-2\\">
    <nav role=\\"navigation\\" aria-label=\\"Pagination Navigation\\" id=\\"pagiElem\\">
      <ul class=\\"v-pagination theme--light\\">
        <li><button disabled=\\"disabled\\" type=\\"button\\" aria-label=\\"Previous page\\" class=\\"v-pagination__navigation v-pagination__navigation--disabled\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-left theme--light\\"></i></button></li>
        <li><button type=\\"button\\" aria-current=\\"true\\" aria-label=\\"Current Page, Page 1\\" class=\\"v-pagination__item v-pagination__item--active primary\\">1</button></li>
        <li><button type=\\"button\\" aria-label=\\"Goto Page 2\\" class=\\"v-pagination__item\\">2</button></li>
        <li><button type=\\"button\\" aria-label=\\"Goto Page 3\\" class=\\"v-pagination__item\\">3</button></li>
        <li><button type=\\"button\\" aria-label=\\"Goto Page 4\\" class=\\"v-pagination__item\\">4</button></li>
        <li><button type=\\"button\\" aria-label=\\"Goto Page 5\\" class=\\"v-pagination__item\\">5</button></li>
        <li><button type=\\"button\\" aria-label=\\"Goto Page 6\\" class=\\"v-pagination__item\\">6</button></li>
        <li><button type=\\"button\\" aria-label=\\"Goto Page 7\\" class=\\"v-pagination__item\\">7</button></li>
        <li><button type=\\"button\\" aria-label=\\"Goto Page 8\\" class=\\"v-pagination__item\\">8</button></li>
        <li><button type=\\"button\\" aria-label=\\"Goto Page 9\\" class=\\"v-pagination__item\\">9</button></li>
        <li><button type=\\"button\\" aria-label=\\"Goto Page 10\\" class=\\"v-pagination__item\\">10</button></li>
        <li><button type=\\"button\\" aria-label=\\"Next page\\" class=\\"v-pagination__navigation\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate mdi mdi-chevron-right theme--light\\"></i></button></li>
      </ul>
    </nav>
    <div class=\\"v-input v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light v-text-field\\">
      <div class=\\"v-input__control\\">
        <div class=\\"v-input__slot\\">
          <div class=\\"v-text-field__slot\\"><label for=\\"input-21\\" class=\\"v-label v-label--active theme--light\\" style=\\"left: 0px; position: absolute;\\">Items per page</label><input id=\\"input-21\\" readonly=\\"readonly\\" type=\\"number\\"></div>
        </div>
        <div class=\\"v-text-field__details\\">
          <div class=\\"v-messages theme--light\\">
            <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
